### PR TITLE
mcs, client: support returning ru statistics in Get/ListResourceGroup response

### LINKED ISSUE #7616: Support persisting ru consumption statistics data and return ru statistics in get response
## Development Task
- Persist ru consumption data in resource group meta
- Change the Get/ListResourceGroup interface to allow return ru consumption statistics

The new interface is used in pingcap/tidb#49318 to support new system table `mysql. request_unit_by_group`

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: Close #7616

### What is changed and how does it work?

```
- persist resource group ru consumption in meta storage
- Add optional param `with_stats` to allow GetResourceGroup/ListResourceGroups return ru consumption data.
```

### Check List

Tests

- Unit test
- Integration test

Code changes

- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/resourcemanager/server/resource_group.go [hunk 1]
```diff
@@ -168,6 +177,11 @@ func (rg *ResourceGroup) IntoProtoResourceGroup() *rmpb.ResourceGroup {
 			RunawaySettings:    rg.Runaway,
 			BackgroundSettings: rg.Background,
 		}
+
+		if rg.RUConsumption != nil {
+			consumption := *rg.RUConsumption
+			group.RUStats = &consumption
```
Comment (nolouch) [inline pkg/mcs/resourcemanager/server/resource_group.go:183]: peresit consumption both in `ResourceGroup.persistSettings` and `SaveResourceGroupStates`? I think we can only persist once in GroupState.

Comment (glorv) [reply pkg/mcs/resourcemanager/server/resource_group.go:183]: This is the side effect that both grpc and persistSettings uses `IntoProtoResourceGroup` to serialize resource group. The RUStats in group setting is not used.

Comment (nolouch) [reply pkg/mcs/resourcemanager/server/resource_group.go:183]: So is `RuStats` useless? can we remove it from proto.ResourceGroup?

Comment (glorv) [reply pkg/mcs/resourcemanager/server/resource_group.go:183]: `RuStats` is used in the grpc response

**Review by nolouch — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -57,12 +57,17 @@ require (
...
-	golang.org/x/tools v0.6.0
-	google.golang.org/grpc v1.54.0
+	golang.org/x/time v0.3.0
+	golang.org/x/tools v0.10.0
+	google.golang.org/grpc v1.59.0
 	gotest.tools/gotestsum v1.7.0
 )
 
+require (
+	google.golang.org/genproto/googleapis/api v0.0.0-20231002182017-d307bd883b97 // indirect
```
Comment (CabinfeverB) [inline go.mod:67]: need format

Comment (glorv) [reply go.mod:67]: This is already formatted by `go mod tidy`

Comment (glorv) [reply go.mod]: done

**Review by CabinfeverB — APPROVED** [review]

File: client/resource_manager_client.go [hunk 3]
```diff
@@ -49,8 +49,8 @@ var ControllerConfigPathPrefixBytes = []byte(controllerConfigPathPrefix)
 
 // ResourceManagerClient manages resource group info and token request.
 type ResourceManagerClient interface {
-	ListResourceGroups(ctx context.Context) ([]*rmpb.ResourceGroup, error)
-	GetResourceGroup(ctx context.Context, resourceGroupName string) (*rmpb.ResourceGroup, error)
+	ListResourceGroups(ctx context.Context, opts ...GetResourceGroupOption) ([]*rmpb.ResourceGroup, error)
```
Comment (bufferflies) [inline client/resource_manager_client.go:52]: how about add new option like `resource_group_name`. It should return the given resource group information .

Comment (glorv) [reply client/resource_manager_client.go:52]: There is a separate  function `GetResourceGroup`

File: pkg/mcs/resourcemanager/server/manager.go [hunk 4]
```diff
@@ -338,12 +338,10 @@ func (m *Manager) persistResourceGroupRunningState() {
 	for idx := 0; idx < len(keys); idx++ {
 		m.RLock()
 		group, ok := m.groups[keys[idx]]
-		m.RUnlock()
 		if ok {
-			m.Lock()
 			group.persistStates(m.storage)
-			m.Unlock()
 		}
+		m.RUnlock()
```
Comment (bufferflies) [inline pkg/mcs/resourcemanager/server/manager.go:344]: why needs to change this?

**Review by bufferflies — APPROVED** [review]

**Pull request merged**